File: 1509-minimum-difference-between-largest-and-smallest-value-in-three-moves/1509-minimum-difference-between-largest-and-smallest-value-in-three-moves.py
class Solution:
    def minDifference(self, nums: List[int]) -> int:
        if len(nums) <= 3: return 0
        
        n = len(nums) - 1
        nums.sort()

        # manually check all 4 combinations
        minDiff = min(nums[n]-nums[3], nums[n-1]-nums[2], nums[n-2]-nums[1], nums[n-3]-nums[0])
        return minDiff

File: 1509-minimum-difference-between-largest-and-smallest-value-in-three-moves/1509-minimum-difference-between-largest-and-smallest-value-in-three-moves.java
class Solution {
    public int minDifference(int[] nums) {
        if(nums.length <= 3) return 0;
        
        var n = nums.length-1;
        Arrays.sort(nums);

        // manually check all 4 combinations
        var minDiff = nums[n]-nums[3];
        minDiff = Math.min(minDiff, nums[n-1]-nums[2]);
        minDiff = Math.min(minDiff, nums[n-2]-nums[1]);
        minDiff = Math.min(minDiff, nums[n-3]-nums[0]);

        return minDiff;
    }
}